src/Uno.UWP/Storage/Pickers/FileOpenPicker.Android.cs

```diff
@@ -42,15 +52,30 @@ internal static bool TryHandleIntent(Intent intent, Result resultCode)
 			return true;
 		}
 
-		private async Task<StorageFile?> PickSingleFileTaskAsync(CancellationToken token)
+		private Task<StorageFile?> PickSingleFileTaskAsync(CancellationToken token)
 		{
-			var files = await PickFilesAsync(false, token);
-			return files.Count == 0 ? null : files[0];
+			var tcs = new TaskCompletionSource<StorageFile?>();
+			NativeDispatcher.Main.Enqueue(async () =>
+			{
+				var files = await PickFilesAsync(false, token);
+				var file = files.Count > 0 ? files[0] : null;
+
+				tcs.TrySetResult(file);
```

Comment:
We should try/catch and call `TrySetException` as well, same for the others.

Reply:
Added!